Sources/Caching/PostedTransactionCache.swift

```diff
@@ -32,15 +35,43 @@ final class PostedTransactionCache: PostedTransactionCacheType {
     }
 
     func savePostedTransaction(_ transaction: StoreTransactionType) {
+        RCIntegrationTestAssertNotMainThread()
+
+        Logger.debug(Strings.purchase.saving_posted_transaction(transaction))
+
         self.deviceCache.update(key: CacheKey.transactions,
-                                default: Set<String>()) { transactions in
+                                default: StoredTransactions()) { transactions in
             transactions.insert(transaction.transactionIdentifier)
         }
     }
 
     func hasPostedTransaction(_ transaction: StoreTransactionType) -> Bool {
-        let transactions: StoredTransactions = self.deviceCache.value(for: CacheKey.transactions) ?? []
-        return transactions.contains(transaction.transactionIdentifier)
+        RCIntegrationTestAssertNotMainThread()
+
+        return self.storedTransactions.contains(transaction.transactionIdentifier)
+    }
+
+    func unpostedTransactions<T: StoreTransactionType>(in transactions: [T]) -> [T] {
+        RCIntegrationTestAssertNotMainThread()
+
+        return Self.unpostedTransactions(in: transactions, with: self.storedTransactions)
+    }
+
+    // MARK: -
+
+    private var storedTransactions: StoredTransactions {
+        return self.deviceCache.value(for: CacheKey.transactions) ?? []
```

Comment:
Kinda unrelated to this PR but I was wondering, do we clear the cache values at any point? In android we clear them when we have a token but not a corresponding active purchase anymore (so the purchase has expired). We perform this check on every app foreground.

Reply:
Done in #3023 and rebased this.